### ISSUE TITLE
POST /api/v1/videos/{id}/views return 400

### ISSUE THREAD

**Issue by kontrollanten** [Chocobozzz/PeerTube#6285]
### Describe the current behavior

In our nginx access log we see a lot of the following errors.

```
[21/Mar/2024:16:58:04 +0000] "POST /api/v1/videos/100dea84-c1d4-49f9-826e-3bb50512b7e4/views HTTP/1.1" 400 207 "/w/2YYZTvhDe4hXyxUWZhgwoY" "Mozilla/5.0 (Linux; Android 12; T431D Build/SP1A.210812.016) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.6261.119 Mobile Safari/537.36" "82.128.208.12" rt=0.005 ua="127.0.0.1:9000" us="400" ut="0.004" ul="207" cs=-
[21/Mar/2024:16:58:09 +0000] "POST /api/v1/videos/100dea84-c1d4-49f9-826e-3bb50512b7e4/views HTTP/1.1" 400 207 "/w/2YYZTvhDe4hXyxUWZhgwoY" "Mozilla/5.0 (Linux; Android 12; T431D Build/SP1A.210812.016) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.6261.119 Mobile Safari/537.36" "82.128.208.12" rt=0.001 ua="127.0.0.1:9000" us="400" ut="0.000" ul="207" cs=-
 [21/Mar/2024:16:58:10 +0000] "POST /api/v1/videos/100dea84-c1d4-49f9-826e-3bb50512b7e4/views HTTP/1.1" 400 207 "/w/2YYZTvhDe4hXyxUWZhgwoY" "Mozilla/5.0 (Linux; Android 12; T431D Build/SP1A.210812.016) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.6261.119 Mobile Safari/537.36" "82.128.208.12" rt=0.003 ua="127.0.0.1:9000" us="400" ut="0.004" ul="207" cs=-
```

We can't see any warnings or errors from Peertube related to these requests.

### Steps to reproduce

1.
2.
3.

### Describe the expected behavior

I assume that the reason for 400 returns is this line https://github.com/Chocobozzz/PeerTube/blob/v6.0.3/server/core/middlewares/validators/videos/video-view.ts#L45

It would be great to log a warning to give more information about why the time is invalid.

### Additional information

* PeerTube instance:
  * Version: 6.0.3
  * NodeJS version: 18.19.1

**Comment by kontrollanten** [comment]
### Describe the current behavior

Currently the client can send float value to POST /api/v1/videos/{id}/views, which isn't accepted by the API due to validators.isInt check 

https://github.com/Chocobozzz/PeerTube/blob/01e4fd067ea4fee30eb53d0e596f6da2244990bb/server/core/middlewares/validators/videos/video-view.ts#L36

https://github.com/Chocobozzz/PeerTube/blob/01e4fd067ea4fee30eb53d0e596f6da2244990bb/server/core/helpers/custom-validators/misc.ts#L71

isInt returns false when passing a float value:
```
validator.isInt('13.12123123')
false
```

The client code that's sending the float is probably this:

https://github.com/Chocobozzz/PeerTube/blob/01e4fd067ea4fee30eb53d0e596f6da2244990bb/client/src/assets/player/shared/peertube/peertube-plugin.ts#L254-L258


### Describe the expected behavior

Either the client should send an int or the API should be more forgiving and converting float to int.

### Additional information

* PeerTube instance:
  * URL:
  * Version: 6.0.3
  * NodeJS version:
  * Ffmpeg version:

* Browser name, version and platforms on which you could reproduce the bug:
* Link to browser console log if relevant:
* Link to server log if relevant (`journalctl` or `/var/www/peertube/storage/logs/`):

**Comment by Chocobozzz** [comment]
> Currently the client can send float value to POST /api/v1/videos/{id}/views, which isn't accepted by the API due to validators.isInt check

Are you sure? In you example I don't see where `startTime` can be a float, and other `notifyUserIsWatching` calls use `Math.floor` (like https://github.com/Chocobozzz/PeerTube/blob/develop/client/src/assets/player/shared/peertube/peertube-plugin.ts#L350)

**Comment by kontrollanten** [comment]
Yes, at least we see that the REST API receives floats from the web client.

In peertube-plugin.ts `startTime` is defined in two places:

https://github.com/Chocobozzz/PeerTube/blob/01e4fd067ea4fee30eb53d0e596f6da2244990bb/client/src/assets/player/shared/peertube/peertube-plugin.ts#L243

https://github.com/Chocobozzz/PeerTube/blob/01e4fd067ea4fee30eb53d0e596f6da2244990bb/client/src/assets/player/shared/peertube/peertube-plugin.ts#L439

The first one is using timeToInt for a reason I guess? I haven't dug deep into this, I've just noticed we receive floats from the web client in our error logs.

**Comment by Chocobozzz** [comment]
Thanks, then I think https://github.com/Chocobozzz/PeerTube/commit/ca889dbbb8bdb6dac3e3585093b5603f20e607f0 will fix the issue. Don't hesitate to comment if you still notice bad requests after applying this patch (don't forget some web browser may still have the old client in cache/as pwa so the user agent can help)
